controllers/nodefeaturediscovery_resources.go

```diff
@@ -69,34 +82,65 @@ func filePathWalkDir(root string) ([]string, error) {
 	return files, err
 }
 
+// getAssetsFrom takes a path as an input and grabs all of the
+// file names in that path, then returns a list of the manifests
+// it found in that path.
 func getAssetsFrom(path string) []assetsFromFile {
 
+	// manifests is a list type where each item in the list
+	// contains the contents of a given asset (manifest)
 	manifests := []assetsFromFile{}
 	assets := path
+
+	// For the given path, find a list of all the files
 	files, err := filePathWalkDir(assets)
 	if err != nil {
 		panic(err)
 	}
+
+	// For each file in the 'files' list, read the file
+	// and store its contents in 'manifests'
 	for _, file := range files {
+
+		// Read the file and return its contents in 'buffer'
 		buffer, err := ioutil.ReadFile(file)
+
+		// If we have an error, then something unexpectedly went
+		// wrong when reading the file's contents
 		if err != nil {
 			panic(err)
 		}
+
+		// If the reading goes smoothly, then append the buffer
+		// (the file's contents) to the list of manifests
 		manifests = append(manifests, buffer)
 	}
 	return manifests
 }
 
 func addResourcesControls(path string) (Resources, controlFunc) {
+
+	// res is a Resources object that contains information about
+	// a given manifest, such as the Namespace and ServiceAccount
+	// being used
 	res := Resources{}
+
+	// ctrl is a controlFunc object that contains a function
+	// that returns information about the status of a resource
+	// (i.e., Ready or NotReady)
 	ctrl := controlFunc{}
 
+	// Get the list of manifests from the given path
 	manifests := getAssetsFrom(path)
 
+	// s and reg are used later on to parse the manifest YAML
 	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme,
 		scheme.Scheme)
 	reg, _ := regexp.Compile(`\b(\w*kind:\w*)\B.*\b`)
 
+	// For each manifest, find its kind, then append the appropriate
+	// function (e.g., 'Namespace' or 'Role') to ctrl so that the
+	// Namespace, Role, etc. can be parsed
```

Comment:
I'd suggest to drop this and e.g. move part of it before the `switch` statement, e.g.

```
// Appednd the appropriate control function depending on the kind
```

or something. WDYT?